packages/seo-integration/src/google-preview/containers/google-preview-container.js

```diff
@@ -40,14 +40,17 @@ const useBaseUrl = ( permalink, slug ) => {
 /**
  * Handles known data for a Google preview component.
  *
- * @param {JSX.Element} as A Google preview component.
- * @param {Object} restProps Props to pass to the Google preview component, that are unhandled by this container.
+ * @param {JSX.Element} as      A Google preview component.
+ * @param {Object} restProps    Props to pass to the Google preview component, that are unhandled by this container.
  *
  * @returns {JSX.Element} A wrapped Google preview component.
  */
 const GooglePreviewContainer = ( { as: Component, ...restProps } ) => {
-	const title = useSelect( select => select( SEO_STORE_NAME ).selectSeoTitle() );
-	const description = useSelect( select => select( SEO_STORE_NAME ).selectMetaDescription() );
+	const titleTemplate = useSelect( select => select( SEO_STORE_NAME ).selectTitleTemplate() );
+	const descriptionTemplate = useSelect( select => select( SEO_STORE_NAME ).selectDescriptionTemplate() );
+	// Assign the templates as the fallback for SEO title and description inside snippet editor if the data from the store is empty.
+	const title = useSelect( select => select( SEO_STORE_NAME ).selectSeoTitle() ) || titleTemplate;
+	const description = useSelect( select => select( SEO_STORE_NAME ).selectMetaDescription() ) || descriptionTemplate;
```

Comment:
The slice for title and description templates are added, where the initial state for Classic editor were provided with the values retrieved from the window object. This approach is preferred since we don't want to access the data from window object in this package (if it's really avoidable). We want to have a more platform-agnostic package, and not all platform might have that data in the window.